File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function GCD(a,b) {
    while (true) {
        if (a % b === 0) {
            return b
        }
        let tmp = a % b
        a = b
        b = tmp
    }
}

function solution(arr) {
    let a,b
    let arr_sort = arr.sort((a,b) => a-b)
    a = arr_sort.pop()
    while (arr_sort.length !== 0) {
        b = arr_sort.pop()
        let tmp = GCD(a,b)
        a = a * b / GCD(a,b)
        }
    return a;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def GCD(a,b):
    while True:
        if a % b == 0:
            return b
        tmp = a % b
        a = b
        b = tmp

def solution(arr):
    answer = 0
    sort_arr = sorted(arr)
    a = sort_arr.pop()
    while sort_arr:
        b = sort_arr.pop()
        tmp = a * b
        a = tmp / GCD(a,b)
    return a 